### PR TITLE
IR-439: Update incident-reporting API endpoints

### PR DESCRIPTION
**Pull request by ushkarev**
* the create and update report endpoints
    * return reports, not their respective inputs
    * need to convert dates in their responses
    * can use stronger date typing on their inputs
* new methods to list, add, update and delete:
    * staff involvement
    * prisoner involvement
    * correction requests
* new methods to list, add and delete questions

### CODE REVIEW

**Review by Mjwillis — APPROVED** [review]
LGTM

File: server/data/incidentReportingApi.ts [hunk 1]
```diff
@@ -423,20 +423,38 @@ export class IncidentReportingApi extends RestClient {
...
 }
 
-class RelatedObjects<ResponseType> {
+type AddStaffInvolvement = {
+  staffUsername: string
+  staffRole: StaffRole
+  comment?: string
+}
+
+type AddPrisonerInvolvement = {
```
Comment (xoen) [inline server/data/incidentReportingApi.ts]: Should these also have the `*Request` suffix?

Comment (ushkarev) [reply server/data/incidentReportingApi.ts]: yes good idea. will add to all type. (there are more in subsequent commits)

Comment (ushkarev) [reply server/data/incidentReportingApi.ts]: blindly adding a `Request` suffixes has resulted in “AddQuestionWithResponsesRequest” … while it follows the pattern, does it sound too stupid?

File: server/data/incidentReportingApi.ts [hunk 2]
```diff
@@ -473,6 +473,13 @@ export class IncidentReportingApi extends RestClient {
     })
     return questions.map(convertQuestionDates)
   }
+
+  async deleteLastQuestionWithResponses(reportId: string): Promise<Question[]> {
```
Comment (xoen) [inline server/data/incidentReportingApi.ts:485]: perhaps a slightly clearer name for this method could be `deleteLastQuestionAndItsResponses()`...more accurate description of what's doing, may help people not-so familiar with the API.

...but maybe pedantic?

File: server/data/incidentReportingApi.ts (hunk 1 above)
Comment (xoen) [reply server/data/incidentReportingApi.ts]: a bit yeah 😅 - especially as there is a mixture or domain concept and HTTP concept in there 😬

**Review by xoen — APPROVED** [review]
nice work :)

**Pull request merged**